streams/runners/storm/runtime/src/main/java/com/hortonworks/streamline/streams/runtime/storm/testing/TestRunProcessorBolt.java

```diff
@@ -35,8 +35,10 @@ public TestRunProcessorBolt(BaseRichBolt processorBolt, String eventLogFilePath)
 
     @Override
     public void prepare(Map map, TopologyContext topologyContext, OutputCollector outputCollector) {
-        EventLoggingOutputCollector collector = new EventLoggingOutputCollector(topologyContext,
-                outputCollector, TestRunEventLogger.getEventLogger(eventLogFilePath));
+        EventCorrelatingOutputCollector collector = new EventCorrelatingOutputCollector(topologyContext,
```

Comment:
Seems the correlation information is added only in test mode? We may optionally need this information for sampling and event correlation in runtime. Have you thought about how we may be able to reuse this?

Reply:
One of possible way is wrapping spouts and bolts and injecting decorated output collector, as same as what we are doing with test mode. 

The issue is how we log events in distributed topology workers. 
If we directly store it to Streamline DB, all the topology workers should be able to access the DB, which also means we need to include appropriate driver library to the topology jar as well.
If we create an API in Streamline and let workers call it, though we need to optimize API requests, it would work in unsecured environment. But we should verify it will also work in secured environment.

Reply:
Ok assuming decorated output collector should work. As long as we are able to inject the information in the event header its fine. We can route it to an internal event logger processor which store the events in a db/hbase etc and expose api for UI. The details needs to be figured out when we implement event correlation for run time.